src/cross_toml.rs

```diff
@@ -67,6 +101,77 @@ impl CrossToml {
         Ok((cfg, unused))
     }
 
+    /// Merges another [`CrossToml`] into `self` and returns a new merged one
+    ///
+    /// # Merging of `targets`
+    /// The `targets` entries are merged based on the [`Target`] keys.
+    /// If a [`Target`] key is present in both configs, the [`CrossTargetConfig`]
+    /// in `other` overwrites the one in `self`.
+    ///
+    /// # Merging of `build`
+    /// The `build` fields ([`CrossBuildConfig`]) are merged based on their sub-fields.
+    /// A field in the [`CrossBuildConfig`] will only overwrite another if it contains
+    /// a value, i.e. it is not `None`.
+    pub fn merge(self, other: CrossToml) -> Result<CrossToml> {
+        type ValueMap = serde_json::Map<String, serde_json::Value>;
+
+        fn to_map<S: Serialize>(s: S) -> Result<ValueMap> {
+            if let Some(obj) = serde_json::to_value(s)?.as_object() {
+                Ok(obj.to_owned())
+            } else {
+                panic!("Failed to serialize CrossToml as object");
```

Comment:
This should not panic, but return a result
```suggestion
                eyre::bail!("failed to serialize CrossToml as object")
```

note also lowercase "f"